src/Agent.Listener/JobDispatcher.cs

```diff
@@ -926,6 +926,11 @@ public bool Cancel(TimeSpan timeout)
                                 timeout = TimeSpan.FromSeconds(60);
                             }
 
+                            if (timeout.TotalMinutes > 35790)
+                            {
```

Comment:
I think it makes sense to add some sort of a warning here instead of just changing the timeout value silently

Reply:
@DaniilShmelev good point, thank you.
I will add the same changes to ADO as well, so in the Agent it will look like an additional check and should not be changed silently. 
Also, I think it would be better to just update the documentation to **cancelTimeoutInMinutes** option [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#timeouts) rather than add additional logic for the warnings.